lib/request.js

```diff
@@ -14,37 +12,44 @@ const version = require("../package.json").version;
  * @returns {Promise}
  */
 function request(baseConfig, method, depth, xml) {
-    const config = _.assign({}, baseConfig);
-
-    config.headers = _.assign({}, baseConfig.headers);
+    const config = {
+        headers: baseConfig.headers,
+        body: xml,
+        method,
+        'headers["Content-length"]': xml.length,
+        'headers["Depth"]': depth,
+        ...baseConfig
+    };
 
     if (!baseConfig.headers || "User-Agent" in baseConfig.headers === false) {
         config.headers["User-Agent"] = "scrapegoat/" + version;
     }
 
     if (baseConfig.auth) {
-        config.auth = _.assign({}, baseConfig.auth);
-        config.auth.sendImmediately = "sendImmediately" in baseConfig.auth ? baseConfig.auth.sendImmediately : false;
+        config.auth = {
+            sendImmediately:
+                "sendImmediately" in baseConfig.auth ?
+                    baseConfig.auth.sendImmediately :
+                    false,
+            ...baseConfig.auth
```

Comment:
We should get the same result as default with spread: 

```
config.auth = {
   sendImmediately: false, //default
   ...baseConfig.auth //overwrites
}
```

That's basically what the expression `"sendImmediately" in baseConfig.auth ? baseConfig.auth.sendImmediately : false,` does, right?